spec/models/search_spec.rb

```diff
@@ -0,0 +1,297 @@
+    end
+
+    context "when user last name is Burden-Hall" do
+      let(:query) { "Burden-Hall" }
+
+      context "with prior authority application" do
+        let(:prepare) { build(:submission, :with_pa_version, defendant_name: "Jack Burden-Hall").tap(&:save) }
+
+        it "returns the record" do
+          expect(search).to eq([prepare.id])
+        end
+
+        context "with additional partial name matches" do
+          let(:prepare) do
+            [
+              build(:submission, :with_pa_version, defendant_name: "Jack Burden-Hall").tap(&:save),
+              build(:submission, :with_pa_version, defendant_name: "Jim Burden").tap(&:save),
+              build(:submission, :with_pa_version, defendant_name: "James Hall").tap(&:save),
+              build(:submission, :with_pa_version, defendant_name: "jack Ball").tap(&:save),
+            ]
+          end
+
+          it "only matches when both name parts are present" do
+            expect(search).to eq([prepare[0].id])
+          end
+        end
+
+        context "when searching on the first part of last name only" do
+          let(:query) { "Burden" }
+
+          it "returns the record" do
+            expect(search).to eq([prepare.id])
+          end
+        end
+
+        context "when searching on the last part of last name only" do
+          let(:query) { "Hall" }
+
+          it "returns the record" do
+            expect(search).to eq([prepare.id])
+          end
+        end
+      end
+
+      context "with non-standard magistrate application" do
+        let(:prepare) { build(:submission, :with_nsm_version, defendant_name: "Jack Burden-Hall").tap(&:save) }
+
+        it "returns the record" do
+          expect(search).to eq([prepare.id])
+        end
+      end
+    end
+
+    context "when firm name is Wonder-1984" do
+      let(:query) { "Wonder-1984" }
+
+      let(:prepare) { build(:submission, firm_name: "Wonder-1984").tap(&:save) }
+
+      it "returns the record" do
+        expect(search).to eq([prepare.id])
+      end
+
+      context "with additional partial name matches" do
+        let(:prepare) do
+          [
+            build(:submission, firm_name: "Wonder-1984").tap(&:save),
+            build(:submission, firm_name: "Wonder").tap(&:save),
+            build(:submission, firm_name: "Hope-1984").tap(&:save),
+            build(:submission, firm_name: "jack Ball").tap(&:save),
+          ]
+        end
+
+        it "only matches when both name parts are present" do
+          expect(search).to eq([prepare[0].id])
+        end
+      end
+
+      context "when searching on the first part of firm name only" do
+        let(:query) { "Wonder" }
+
+        it "returns the record" do
+          expect(search).to eq([prepare.id])
+        end
+      end
+
+      context "when searching on the last part of firm name only" do
+        let(:query) { "1984" }
+
+        it "returns the record" do
+          skip "does not match as expects `-1984`"
```

Comment:
should this work?

Reply:
see what comes out with meeting with Mati - imo probably but would accept it for now (as long as people can search firms against key words, I doubt there will be an instance where people are anything more than mildly inconvenienced from only being able to filter via the keywords and not the number in this edge case)